### PR TITLE
Feature/us19 login part 1

### PR DESCRIPTION
**Pull request by ufelgen**
following [this US](https://github.com/ufelgen/capstone-project/issues/40) I added a login using [next-auth](https://next-auth.js.org/getting-started/example). if you sign in with GitHub, the main page is rendered and your username is shown. in the vercel preview, you can sign in with username and password: "extreme-beaver" 😉🦫

### CODE REVIEW

File: pages/api/auth/[...nextauth].js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        password: { label: "Password", type: "password" },
+      },
+      async authorize(credentials) {
+        if (
+          credentials.username === "extreme-beaver" &&
+          credentials.password === "extreme-beaver"
+        ) {
+          return {
+            name: "Beaver Test User",
+            email: "test@example.com",
```
Comment (AntjeWalter) [inline pages/api/auth/[...nextauth].js:21]: pretty nice feature!

File: pages/index.js [hunk 2]
```diff
@@ -75,50 +79,68 @@ export default function Home({
...
-            </Fragment>
-          )
+      <LoginSection>
+        {session ? (
+          <>
+            <div>
+              <p>you are signed in as </p>
+              <p>{session?.user?.name}</p>
+            </div>
+            <BiggerActionButton onClick={signOut}>SIGN OUT</BiggerActionButton>
```
Comment (AntjeWalter) [inline pages/index.js:89]: is signOut a function from nextAuth that you don't have to write yourself and it just works like magic? :D

**Review by AntjeWalter — APPROVED** [review]
Since I don't know much about Logins, I can't say a lot about the correctness of your code but it looks good to me and I think I understand the basic functionality. :)

File: pages/index.js (hunk 2 above)
Comment (ufelgen) [reply pages/index.js:89]: yes, it's all background magic that you can just import 🙈

**Pull request merged**